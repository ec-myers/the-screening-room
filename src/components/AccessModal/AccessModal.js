import React from 'react';
import './AccessModal.scss';
import LoginForm from '../LoginForm/LoginForm';
import SignUpForm from '../SignUpForm/SignUpForm';
import { Route, NavLink, Link } from 'react-router-dom';
import { connect } from 'react-redux';

export const AccessModal = ({ user, findUserFavorites }) => {
  return (
    <>
      <Link to='/'>
        <div className='modal-backdrop-white'>
          <div className='modal-backdrop'></div>
        </div>
      </Link>
      <div className='modal-container'>
        <ul className='modal-tabs'>
          <NavLink to='/login' className='form-navlink'>
            <li className='login_tab'>Log in</li>
          </NavLink>
          <NavLink to='/signup' className='form-navlink'>
            <li className='login_tab'>Sign up</li>
          </NavLink>
        </ul>
        <Route exact path='/login' render={() => <LoginForm />} />
        <Route exact path='/signup' render={() => <SignUpForm />} />
      </div>
    </>
  );
};

export const mapStateToProps = user => ({
  user
});

export default connect(mapStateToProps)(AccessModal);
