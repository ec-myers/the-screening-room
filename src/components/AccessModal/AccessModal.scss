.modal-backdrop {
  width: 100vw;
  height: 100vh;
  position: fixed;
  top: 50%;
  left: 50%;
  content: url('https://i.pinimg.com/originals/6a/90/86/6a9086949c9b9451026f0095d97c7c2c.jpg');
  transform: translate3d(-50%, -50%, 0);
  opacity: 0.05;
}

.modal-backdrop-white {
  background: white;
  width: 100vw;
  height: 100vh;
  position: fixed;
  top: 50%;
  left: 50%;
  transform: translate3d(-50%, -50%, 0);
}

.modal-container {
  position: absolute;
  z-index: 100;
  top: 50%;
  left: 50%;
  transform: translate3d(-50%, -50%, 0);
  display: flex;
  flex-direction: column;
  background-color: white;
  border-radius: 1px;
  border: none;
  box-shadow: 0 2px 10px rgba(0, 0, 0, 0.301);
  min-width: 20rem;
  max-width: 60rem;
  width: 500px;
  height: 600px;
  background: linear-gradient(to top left, rgb(139, 228, 255), white, white, white, white);

  li {
    width: 250px;
    list-style-type: none;
    text-align: center;
    font-size: 30px;
    font-family: 'Roboto';
    font-weight: 500;
    padding: 10px 0;
  }

  ul {
    width: 100%;
  }
}

.form-navlink {
  text-decoration: none;
  color: white;
  transition: .15s;

  &:hover {
    background: rgb(0, 0, 0);
    transition: .2s;
  }
}

.active {
  color: rgb(27, 27, 27);
  background: white;

  &:hover {
    background: white;
  }
}

.modal-tabs {
  display: flex;
  justify-content: space-evenly;
  width: 100%;
  background: rgb(41, 41, 41);
}