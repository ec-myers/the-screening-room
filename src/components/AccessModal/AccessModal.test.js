import React from 'react';
import { shallow } from 'enzyme';
import { AccessModal, mapStateToProps } from './AccessModal';

describe('AccessModal', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<AccessModal />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
