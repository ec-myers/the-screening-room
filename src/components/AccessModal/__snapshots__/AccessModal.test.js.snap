// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccessModal should match the snapshot 1`] = `
<Fragment>
  <Link
    to="/"
  >
    <div
      className="modal-backdrop-white"
    >
      <div
        className="modal-backdrop"
      />
    </div>
  </Link>
  <div
    className="modal-container"
  >
    <ul
      className="modal-tabs"
    >
      <NavLink
        className="form-navlink"
        to="/login"
      >
        <li
          className="login_tab"
        >
          Log in
        </li>
      </NavLink>
      <NavLink
        className="form-navlink"
        to="/signup"
      >
        <li
          className="login_tab"
        >
          Sign up
        </li>
      </NavLink>
    </ul>
    <Route
      exact={true}
      path="/login"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/signup"
      render={[Function]}
    />
  </div>
</Fragment>
`;
